var/spack/repos/builtin/packages/vasp/package.py

```diff
@@ -34,18 +35,29 @@ class Vasp(MakefilePackage):
     )
 
     variant("openmp", default=False, description="Enable openmp build")
+    with when("+openmp"):
+        conflicts("^fftw~openmp")
+        conflicts("^amdfftw~openmp")
+        conflicts("^amdblis threads=none")
+        conflicts("^amdblis threads=pthreads")
+        conflicts("^openblas threads=none")
+        conflicts("^openblas threads=pthreads")
 
     variant("scalapack", default=False, description="Enables build with SCALAPACK")
 
     variant("cuda", default=False, description="Enables running on Nvidia GPUs")
+    variant("fftlib", default=False, description="Enables fftlib build")
+    with when("+fftlib"):
+        conflicts("^mkl", msg="fftlib is not supported for Intel MKL")
+        conflicts("@:6.1.1", msg="fftlib support started from 6.2.0")
+        conflicts("~openmp", msg="fftlib is intended to be used with openmp")
```

Comment:
Should the dependency on `fftw-api` now look like this: `depends_on("fftw-api", when="~fftlib")`?  I'm not super familiar with VASP but maybe @fcannini or @alansill knows better (just looking at the contributor history).

Also can MKL still be used with fftlib if MKL's fftw isn't linked?  @alalazo FYI

Reply:
> Should the dependency on fftw-api now look like this: depends_on("fftw-api", when="~fftlib") ?

FFTW can work with or without VASP’s FFLIB layer.

> can MKL still be used with fftlib if MKL's fftw isn't linked ?

If MKL is built/enabled with FFTW wrappers, then it will use VASP’s FFTLIB provided FFTLIB is enabled. But, if MKL is not built/enabled with FFTW wrappers, then it will not use VASP’s FFTLIB even if FFTLIB is enabled. FFTLIB layer can be enabled and built into the VASP binary irrespective of whether a FFT library (FFTW or MKL) use it or not.